specs/core/0_beacon-chain.md

```diff
@@ -2310,6 +2341,60 @@ def process_attester_slashing(state: BeaconState,
         slash_validator(state, index)
 ```
 
+##### Randao reveal slashings
+
+Verify that `len(block.body.randao_reveal_slashings) <= MAX_RANDAO_REVEAL_SLASHINGS`.
+
+For each `randao_reveal_slashing` in `block.body.randao_reveal_slashings`, run the following function:
+
+```python
+def process_randao_reveal_slashing(state: BeaconState,
+                                   randao_reveal_slashing: RandaoRevealSlashing) -> None:
+    """
+    Process ``RandaoRevealSlashing`` operation.
+    Note that this function mutates ``state``.
+    """
+    revealer = state.validator_registry[randao_reveal_slashing.revealer_index]
+    masker = state.validator_registry[randao_reveal_slashing.masker_index]
+    pubkeys = [revealer.pubkey, masker.pubkey]
+    message_hashes = [
+        hash_tree_root(randao_reveal_slashing.epoch),
+        randao_reveal_slashing.mask,
+    ]
+
+    assert randao_reveal_slashing.epoch >= get_current_epoch(state) + RANDAO_SLASHING_EPOCHS
+    assert revealer.slashed is False
+    assert randao_reveal_slashing.epoch not in state.validator_registry[randao_reveal_slashing.revealer_index].exposed_randao_reveals
+
+    assert bls_verify_multiple(
+        pubkeys=pubkeys,
+        message_hashes=message_hashes,
+        signature=randao_reveal_slashing.reveal,
+        domain=get_domain(
+            fork=state.fork,
+            epoch=randao_reveal_slashing.epoch,
+            domain_type=DOMAIN_RANDAO,
+        ),
+    )
+
+    if randao_reveal_slashing.epoch >= get_current_epoch(state) + CUSTODY_PERIOD_TO_RANDAO_PADDING:
```

Comment:
As discussed in person, I'd support just setting CUSTODY_PERIOD_TO_RANDAO_PADDING to equal EPOCHS_PER_CUSTODY_PERIOD for simplicity and added conservatism.

Reply:
So I did this but I noticed there is a problem with this approach: If we do this, then we will have to wait for a whole custody period after the round key has been used to reveal the key (or else the validator would be subject to RANDAO slashing)! So as a compromise between the two I chose CUSTODY_PERIOD_TO_RANDAO_PADDING=2**8 (13 hours)